### PR TITLE
Packaging: Stop and disable service on DEB package removal

### PR DESCRIPTION
**Pull request by MichaIng**
**What is this feature?**

The Grafana server service is now stopped and disabled on package removal.

**Why do we need this feature?**

Currently the service keeps running after package removal, failing at some point due to missing files. And if enabled, the attempted service start on every boot is doomed to fail.

The condition for this code to run applies on package removal only, not on upgrade or reinstall which generally invoke the `prerm` script as well.

If the `invoke-rc.d` command exists, it can be assumed that `update-rc.d` exists as well, as both are part of the same `init-system-helpers` package in dpkg distro repositories.

If neither systemd, nor the init system helper package is installed, the service is not tried to be disabled, since we cannot know a safe method to do so.

Compared to the `postinst` script, `set -e` is skipped here, since it is implied by dpkg maintainer script invocation, and since we do not run any command which is allowed to fail the whole package removal.

**Who is this feature for?**

Everyone who starts and/or enables the service via systemd or SysV will benefit, since the service doesn't need to be disabled manually anymore after package removal, to avoid service startup failures. Disabling isn't even trivial after the service files have been removed already together with the package, links might need to be removed manually.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by MichaIng** [comment]
Same could/should be added for RPM packages as well, but I'm not familiar with such distros. How to stop the service can be derived from the RPM [postinst](https://github.com/grafana/grafana/blob/main/packaging/rpm/control/postinst) script, but not sure how to disable it. Better to be added by someone with RPM packaging experience.

File: packaging/deb/control/prerm
Comment (kminehart) [inline packaging/deb/control/prerm:3]: Is this if statement necessary since this is in the --before-remove script?

**Review by kminehart — CHANGES_REQUESTED** [review]

File: packaging/deb/control/prerm
Comment (kminehart) [inline packaging/deb/control/prerm:1]: I think we would prefer the shebang to look like this: `#!/usr/bin/env sh`

https://unix.stackexchange.com/questions/12736/how-does-usr-bin-env-know-which-program-to-use/12751#12751

Comment (MichaIng) [reply packaging/deb/control/prerm:1]: Usually I'd do so as well. Official Debian repo packages however use `#!/bin/sh` when I checked last. However, both will work reliably. I'll then also align this with the `postinst` script.

File: packaging/deb/control/prerm
Comment (MichaIng) [reply packaging/deb/control/prerm:3]: Yes, it's necessary since `prerm` is called as well on package upgrades for the old package with `upgrade` argument. And we do not want the service to be disabled on upgrades.

But we may want to have it stopped/restarted before/after upgrades? However, something for a new PR in case.

File: packaging/deb/control/prerm
Comment (MichaIng) [reply packaging/deb/control/prerm]: Adjusted on both maintainer scripts.

**Review by DanCech — APPROVED** [review]
Nice work!

**Comment by DanCech** [comment]
@kminehart looks like this should be good to go now, wdyt?

File: pkg/build/packaging/grafana.go
Comment (DanCech) [inline pkg/build/packaging/grafana.go:382]:
```suggestion
	if options.prermSrc != "" {
```
this can never be nil so we can just check it's not empty

**Comment by kminehart** [comment]
Yeah i think so. I'm on vacation until next week but it you wanted to merge it then LGTM.

**Review by kminehart — APPROVED** [review]

**Comment by DanCech** [comment]
Thanks for the contribution @MichaIng !

**Pull request merged**
